packages/contracts/src/chugsplash/hardhat-tools.ts

```diff
@@ -0,0 +1,54 @@
+/* Imports: External */
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+/* Imports: Internal */
+import { computeStorageSlots, getStorageLayout } from './storage'
+import { ChugSplashConfig, parseChugSplashConfig } from './config'
+import {
+  ChugSplashAction,
+  ChugSplashActionBundle,
+  getChugSplashActionBundle,
+} from './actions'
+
+/**
+ * Generates a ChugSplash action bundle from a config file.
+ * @param hre Hardhat runtime environment, used to load artifacts + storage layouts.
+ * @param config Config file to convert into a bundle.
+ * @param env Environment variables to inject into the config file.
+ * @returns Action bundle generated from the parsed config file.
+ */
+export const makeActionBundleFromConfig = async (
+  hre: HardhatRuntimeEnvironment,
+  config: ChugSplashConfig,
+  env: any = {}
+): Promise<ChugSplashActionBundle> => {
+  // Parse the config to replace any template variables.
+  const parsed = parseChugSplashConfig(config, env)
+
+  const actions: ChugSplashAction[] = []
+  for (const [contractName, contractConfig] of Object.entries(
+    parsed.contracts
+  )) {
+    const artifact = hre.artifacts.readArtifactSync(contractConfig.source)
+    const storageLayout = await getStorageLayout(hre, contractConfig.source)
```

Comment:
Do you think it could be possible to pull this asynchronous call out of the critical path in some way? It might make it easier to test this function if it was entirely synchronous. It might require some intermediate representation of the config.

Reply:
I don't think this is particularly easy without adding some ugly code. Source of the `async` nature of this function comes from a single call to [`hre.artifacts.getBuildInfo`](https://github.com/nomiclabs/hardhat/blob/40520e3f05f87b58a6ba0f81ac0904646a7ce3f1/packages/hardhat-core/src/internal/artifacts.ts#L129-L144) specifically [here](https://github.com/ethereum-optimism/optimism/blob/73308063553c56f8d6d17115b1d1c2ca2d9cb00a/packages/contracts/src/chugsplash/storage.ts#L73-L75) in `getStorageLayout`. If we want to make this sync we'd have to duplicate most of the logic of `getBuildInfo` and remove the async calls made within.